File: identityscanner/src/main/java/com/digitify/identityscanner/activities/PermissionAwareActivity.java
package com.digitify.identityscanner.activities;

import android.os.Bundle;

import com.digitify.identityscanner.helpers.PermissionsManager;

import androidx.annotation.Nullable;
import androidx.lifecycle.LifecycleOwner;

public abstract class PermissionAwareActivity extends BaseActivity implements PermissionsManager.OnPermissionGrantedListener {

    private PermissionsManager permissionsManager;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        permissionsManager = new PermissionsManager(this, this, this);
    }

    public PermissionsManager getPermissionsManager() {
        return permissionsManager;
    }

    @Override
    public void onPermissionGranted(String permission) {

    }

    @Override
    public void onPermissionNotGranted(String permission) {

    }

    @Override
    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        permissionsManager.onRequestPermissionsResult(requestCode, permissions, grantResults);
    }
}

File: yapcore/src/main/java/co/yap/yapcore/PermissionAwareActivity.kt
package co.yap.yapcore;

import android.os.Bundle
import androidx.lifecycle.LifecycleOwner
import co.yap.yapcore.helpers.PermissionsManager

abstract class PermissionAwareActivity : BaseActivity(), PermissionsManager.OnPermissionGrantedListener,
    LifecycleOwner {

    private lateinit var permissionsManager: PermissionsManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        permissionsManager = PermissionsManager(this, this, this)
    }

    override fun onPermissionGranted(permission: String?) {
    }

    override fun onPermissionNotGranted(permission: String?) {
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        permissionsManager.onRequestPermissionsResult(requestCode, permissions, grantResults)
    }

    fun isPermissionGranted(permission: String): Boolean {
        return permissionsManager.isPermissionGranted(permission)
    }

    fun requestPermissions() {
        return permissionsManager.requestAppPermissions()
    }
}
